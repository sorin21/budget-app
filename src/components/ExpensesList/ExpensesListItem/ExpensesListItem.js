import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom'
import { Link } from 'react-router-dom';
import numeral from "numeral";
import moment from 'moment';

import { removeExpense } from '../../../actions/expensesActions';


class ExpensesListItem extends Component {
  render() {
    const { id, description, amount, createdAt, note } = this.props.expense;
    return (
      <div>
        <Link to={`/edit/${id}`}>
          <h3>Description: {description}</h3>
        </Link>
        <p>Amount: {numeral(amount).format('0,0.00')}</p>
        <p>Created At: {moment(createdAt).format('MMMM Do, YYYY HH:mm')}</p>
        {note && <p>Note: {note}</p>}
        <button
          onClick={() => {
            this.props.onRemoveExpense(id);
            this.props.history.push('/');
          }}
        >Remove</button>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    expenses: state.expenses
  }
}

const mapDispatchProps = dispatch => {
  return {
    onRemoveExpense: (id) => dispatch(removeExpense({ id }))
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchProps)(ExpensesListItem));