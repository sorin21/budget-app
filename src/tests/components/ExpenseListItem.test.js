import React from 'react';
import { shallow } from 'enzyme';

import expenses from '../../common/expenses';
import { ExpensesListItem } from '../../components/ExpensesList/ExpensesListItem/ExpensesListItem';

test('should render ExpensesListItem component', () => {
  const wrapper = shallow(<ExpensesListItem {...expenses[0]} />);
  expect(wrapper).toMatchSnapshot();
})