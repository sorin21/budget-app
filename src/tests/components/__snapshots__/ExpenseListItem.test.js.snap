// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesListItem component 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      Description: 
      Gum
    </h3>
  </Link>
  <p>
    Amount: 
    195.00
  </p>
  <p>
    Created At: 
    January 1st, 1970 02:00
  </p>
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
